src/tokenizer.rs

```diff
@@ -1229,15 +1229,31 @@ impl<'a> Tokenizer<'a> {
                 // operators
                 '-' => {
                     chars.next(); // consume the '-'
-                    match chars.peek() {
-                        Some('-') => {
-                            chars.next(); // consume the second '-', starting a single-line comment
+
+                    if let Some('-') = chars.peek() {
```

Comment:
would it be possible to have this remain as a match statement, calling `chars.peek()` only once? i.e.
```rust
match chars.peek() {
    Some('-') => { ... }
    Some('>') => { ... }
 }
```